suites/test_view_assembly.py

```diff
@@ -0,0 +1,110 @@
+        # utilities.click_element(self.driver, By.CSS_SELECTOR, locators.EDIT_METADATA_BUTTON_CSS)
+        utilities.click_element(self.driver, By.XPATH, locators.ADD_METADATA_BUTTON_XPATH)
+        check_that("Edit metadata modal title",
+                   utilities.get_text(self.driver, By.CSS_SELECTOR, locators.EDIT_METADATA_MODAL_TITLE_CSS),
+                   contains_string(constants.edit_metadata_modal_title))
+        display_module_page.fill_edit_metadata_form(self.driver, constants.product_name, constants.product_version,
+                                                    constants.use_case, constants.url_fragment)
+        check_that("Success message displayed",
+                   utilities.get_text(self.driver, By.CSS_SELECTOR, locators.UPDATE_SUCCESS_MESSAGE_CSS),
+                   contains_string(constants.success_message))
+        check_that("Product name reflected on displayed module page",
+                   utilities.get_text(self.driver, By.CSS_SELECTOR, locators.PRODUCT_INFO_CSS),
+                   contains_string(constants.product_name))
+        check_that("Product version reflected on displayed module page",
+                   utilities.get_text(self.driver, By.CSS_SELECTOR, locators.PRODUCT_INFO_CSS),
+                   contains_string(constants.product_version))
+        utilities.wait(2)
+
+    @lcc.test("Verify that user is able to publish module successfully")
+    def publish_assembly(self):
+        # utilities.click_element(self.driver, By.LINK_TEXT, "Search")
+        # search_page.search_for_module_and_click(self.driver, constants.assembly_to_be_published)
+        utilities.wait(5)
+        utilities.click_element(self.driver, By.CSS_SELECTOR, locators.MODULE_DISPLAY_PUBLISH_BUTTON_CSS)
+        check_that("URL", self.driver.current_url,
+                   contains_string(url + constants.module_display_page_path_after_published))
+        check_that("Button", utilities.get_text(
+            self.driver, By.CSS_SELECTOR, locators.MODULE_DISPLAY_PREVIEW_BUTTON_CSS), contains_string("Preview"))
+        check_that("Button", utilities.get_text(
+            self.driver, By.CSS_SELECTOR, locators.MODULE_DISPLAY_PUBLISH_BUTTON_CSS), contains_string("Unpublish"))
+        # Add a check that the Published column contains some Published time.
+
+    @lcc.test("Verify contents of assembly preview in pantheon")
+    def preview_assembly(self):
+        # Get list of modules included
+        modules_included = utilities.find_elements_by_css_selector(self.driver, locators.MODULES_INCLUDED_LIST_CSS)
+        modules_count = len(modules_included)
+        module_titles = []
+        for i in range(modules_count):
+            module_titles.append(modules_included[i].text)
+        utilities.click_element(self.driver, By.CSS_SELECTOR, locators.MODULE_DISPLAY_PREVIEW_BUTTON_CSS)
+        print("Clicked preview")
+        utilities.wait(5)
+        utilities.switch_to_latest_tab(self.driver)
```

Comment:
please handle the switching between tabs in try,catch finally blocks, in case of a failure check, all the further tests fail as the tab does not switch to the first one then.